mooncake-store/include/segment.h

```diff
@@ -0,0 +1,183 @@
+#pragma once
+
+#include <boost/functional/hash.hpp>
+#include <ostream>
+#include <shared_mutex>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+#include <variant>
+#include <vector>
+
+#include "allocation_strategy.h"
+#include "allocator.h"
+#include "types.h"
+
+namespace mooncake {
+/**
+ * @brief Status of a mounted segment in master
+ */
+enum class SegmentStatus {
+    UNDEFINED = 0,  // Uninitialized
```

Comment:
We can probably simplify this to just two statuses, since 'undefined' shouldn't ever show up anyway